### ISSUE TITLE
Not Working!!

### ISSUE THREAD

**Issue by tushar-wq** [kallqvist/skillshare-downloader#68]
Even after changing cookies many times, it does not work. It downloads the first video and then shows an exception 
"Exception: Failed to read video ID from data" 

Can anyone please see what the issue is? I am using the cookies for my premium account. It was working 4 days ago but when I try today it does not seem to work :(

**Comment by trixal** [comment]
Yes, it can't work. A change has been made. I already know how to do it gradually, but I haven't been able to find out the key here how each video changes the first original bc: xxx, the subsequent links are also encrypted 'video_hashed_id'. Does anyone want to participate in the survey? My method works, but it is not suitable for screwing.

**Comment by ForsakenNGS** [comment]
I'm down, just tell me what you need ;)

**Comment by KClaassen** [comment]
Same, happy to have a look :) What workaround are you currently using?

**Comment by BitWizCoder-ol** [comment]
Same error. "Exception: Failed to read video ID from data"
i used to use software call allavsoft. but that software need actual login cant use cookies.
waiting if for any solution to the problem.
Thanks.

**Comment by DarkSuniuM** [comment]
#65 Fixes this.

**Comment by piratesephiroth** [comment]
> #65 Fixes this.

Not anymore. The original script still works though.
It's just that they made the PHPSESSID cookie HttpOnly, so it's hidden from "document.cookie" and that's the only cookie you actually need for downloading the videos. 
Now you must get it using a cookie manager browser extension.

**Comment by Gorgosus** [comment]
Ya, you have to download "EditThisCookie" extension from https://chrome.google.com/webstore/detail/editthiscookie/fngmhnnpilhplaeedifhccceomclgfbg. Then click this extension and set "Choose the preferred export format for cookies" as "semicolon separated" in settings. After that, just export cookie from cookie editior, and paste copied string to python just like in tutorial

**Comment by DarkSuniuM** [comment]
> > #65 Fixes this.
> 
> Not anymore. The original script still works though. It's just that they made the PHPSESSID cookie HttpOnly, so it's hidden from "document.cookie" and that's the only cookie you actually need for downloading the videos. Now you must get it using a cookie manager browser extension.

#65 has nothing to do with cookies, and doesn't change behavior of the script with cookies, That just reads the video id from thumbnail as an alternative method, so...

**Comment by piratesephiroth** [comment]
> > > #65 Fixes this.
> > 
> > 
> > Not anymore. The original script still works though. It's just that they made the PHPSESSID cookie HttpOnly, so it's hidden from "document.cookie" and that's the only cookie you actually need for downloading the videos. Now you must get it using a cookie manager browser extension.
> 
> #65 has nothing to do with cookies, and doesn't change behavior of the script with cookies, That just reads the video id from thumbnail as an alternative method, so...

Yes, I know that, and that thumbnail video id has already changed to a different format, that's why I wrote "not anymore".
The root of the issue has everything to do with cookies. You just need the PHPSESSID one and that's been put out of reach of javascript, so that's the reason people can't access any paid videos when they get the cookies from the javascript console.

**Comment by ogeeDeveloper** [comment]
How do i go about to download videos without this annoying exception error, my internet is slow and i want to download videos

**Comment by piratesephiroth** [comment]
> How do i go about to download videos without this annoying exception error, my internet is slow and i want to download videos

Do as I wrote above, use a cookie manager browser extension and get your PHPSESSID cookie from the skillshare website. You only need that to download videos.
Then add it like:
`cookie = "PHPSESSID=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"`

**Comment by abu3beid** [comment]
> > How do i go about to download videos without this annoying exception error, my internet is slow and i want to download videos
> 
> Do as I wrote above, use a cookie manager browser extension and get your PHPSESSID cookie from the skillshare website. You only need that to download videos. Then add it like: `cookie = "PHPSESSID=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"`

Thank you so much ♥

**Comment by zachshirow** [comment]
thank you, thank you, thank you, thank you, thank you, thank you, thank you, thank you, thank you, thank you,

**Comment by rroyo** [comment]
The PHPSESSID cookie works perfectly, thanks!
Now though the script names the files to their Spanish translation —I'm in Spain.
When using the default cookie, the name of the downloaded file was in English.
Any idea on how to change this behavior?

**Comment by piratesephiroth** [comment]
> The PHPSESSID cookie works perfectly, thanks! Now though the script names the files to their Spanish translation —I'm in Spain. When using the default cookie, the name of the downloaded file was in English. Any idea on how to change this behavior?

Yeah you have to add an "accept_language" header now, otherwise it's gonna use the language based on your location..
You can try my fork, I made it use English and it also saves the files with better names and organizes them in folders.

**Comment by rroyo** [comment]
Thanks for the prompt response. Your fork is working perfectly.

**Comment by ghazikr** [comment]
@piratesephiroth Thank you!!!! I am curious how did the developer figure out that `https://api.skillshare.com/classes/` is the endpoint to call ?  I looked into the network tab but failed to see any calls to that endpoint.

**Comment by 04anubhavv** [comment]
> 

He checked the network activity of the Skillshare mobile app, as the script uses `Skillshare/5.3.0; Android 9.0.1` user agent.
